// nekosapi/v3/types/Tags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tags = void 0;
var Tags;
(function (Tags) {
    Tags[Tags["Illustration"] = 1] = "Illustration";
    Tags[Tags["Girl"] = 2] = "Girl";
    Tags[Tags["Black hair"] = 3] = "Black hair";
    Tags[Tags["Sportswear"] = 4] = "Sportswear";
    Tags[Tags["Sword"] = 5] = "Sword";
    Tags[Tags["Kemonomimi"] = 6] = "Kemonomimi";
    Tags[Tags["Flowers"] = 7] = "Flowers";
    Tags[Tags["Catgirl"] = 8] = "Catgirl";
    Tags[Tags["White hair"] = 9] = "White hair";
    Tags[Tags["Loli"] = 10] = "Loli";
    Tags[Tags["Plants"] = 11] = "Plants";
    Tags[Tags["Blue Hair"] = 12] = "Blue Hair";
    Tags[Tags["Pink hair"] = 13] = "Pink hair";
    Tags[Tags["Purple hair"] = 14] = "Purple hair";
    Tags[Tags["Exposed girl breasts"] = 15] = "Exposed girl breasts";
    Tags[Tags["Exposed anus"] = 16] = "Exposed anus";
    Tags[Tags["Pussy"] = 17] = "Pussy";
    Tags[Tags["Dick"] = 18] = "Dick";
    Tags[Tags["Maid"] = 19] = "Maid";
    Tags[Tags["Beach"] = 20] = "Beach";
    Tags[Tags["Reading"] = 21] = "Reading";
    Tags[Tags["Mountain"] = 22] = "Mountain";
    Tags[Tags["Night"] = 23] = "Night";
    Tags[Tags["Gloves"] = 24] = "Gloves";
    Tags[Tags["Original Style"] = 25] = "Original Style";
    Tags[Tags["Brown Hair"] = 26] = "Brown Hair";
    Tags[Tags["Sunny"] = 27] = "Sunny";
    Tags[Tags["Rain"] = 28] = "Rain";
    Tags[Tags["Shorts"] = 29] = "Shorts";
    Tags[Tags["Weapon"] = 30] = "Weapon";
    Tags[Tags["Bikini"] = 31] = "Bikini";
    Tags[Tags["Ice Cream"] = 32] = "Ice Cream";
    Tags[Tags["Tree"] = 33] = "Tree";
    Tags[Tags["Bunny girl"] = 34] = "Bunny girl";
    Tags[Tags["Dress"] = 35] = "Dress";
    Tags[Tags["Usagimimi"] = 36] = "Usagimimi";
    Tags[Tags["School Uniform"] = 37] = "School Uniform";
    Tags[Tags["Guitar"] = 38] = "Guitar";
    Tags[Tags["Baggy Clothes"] = 39] = "Baggy Clothes";
    Tags[Tags["Wet"] = 40] = "Wet";
    Tags[Tags["Yuri"] = 41] = "Yuri";
    Tags[Tags["Red hair"] = 42] = "Red hair";
    Tags[Tags["Glasses"] = 43] = "Glasses";
    Tags[Tags["Anal"] = 44] = "Anal";
    Tags[Tags["Futanari"] = 45] = "Futanari";
    Tags[Tags["Masturbating"] = 46] = "Masturbating";
    Tags[Tags["Threesome"] = 47] = "Threesome";
    Tags[Tags["Kissing"] = 48] = "Kissing";
    Tags[Tags["Skirt"] = 49] = "Skirt";
    Tags[Tags["Blonde hair"] = 50] = "Blonde hair";
    Tags[Tags["Horsegirl"] = 51] = "Horsegirl";
    Tags[Tags["Boy"] = 52] = "Boy";
})(Tags || (exports.Tags = Tags = {}));
